### PR TITLE
TACO-35 adding mutable tracing info property to the response and request protocols

### PR DESCRIPTION
**Pull request by manimaul**
This PR adds a mutable `TraceInfo` property to the `Request` and `Response` protocols in order to propagate zipkin traces.

PR also uses `Optional<Span>` rather than nullable annotations.

### CODE REVIEW

File: xio-core/src/main/java/com/xjeffrose/xio/http/DefaultFullRequest.java [hunk 1]
```diff
@@ -71,9 +66,21 @@ public Builder host(String host) {
...
+
+    abstract Optional<TraceInfo> httpTraceInfo();
+
+    abstract DefaultFullRequest autoBuild();
+
+    public DefaultFullRequest build() {
+      if (!httpTraceInfo().isPresent()) {
+        httpTraceInfo(new TraceInfo());
+      }
+      streamId(-1);
```
Comment (pdex) [inline xio-core/src/main/java/com/xjeffrose/xio/http/DefaultFullRequest.java]: Please put this back into `builder` as it is a default there and a constant here.

File: xio-core/src/main/java/com/xjeffrose/xio/http/DefaultStreamingRequest.java [hunk 2]
```diff
@@ -66,9 +61,21 @@ public Builder host(String host) {
...
+
+    abstract Optional<TraceInfo> httpTraceInfo();
+
+    abstract DefaultStreamingRequest autoBuild();
+
+    public DefaultStreamingRequest build() {
+      if (!httpTraceInfo().isPresent()) {
+        httpTraceInfo(new TraceInfo());
+      }
+      streamId(-1);
```
Comment (pdex) [inline xio-core/src/main/java/com/xjeffrose/xio/http/DefaultStreamingRequest.java]: same ask here as `DefaultFullRequest`

File: xio-core/src/main/java/com/xjeffrose/xio/http/StreamingRequestData.java [hunk 3]
```diff
@@ -76,10 +74,9 @@ public ByteBuf body() {
 
   // region Traceable
 
-  @Nullable
   @Override
-  public Span traceSpan() {
-    return span;
+  public TraceInfo httpTraceInfo() {
```
Comment (pdex) [inline xio-core/src/main/java/com/xjeffrose/xio/http/StreamingRequestData.java]: Removed @Nullable 👍

File: xio-core/src/main/java/com/xjeffrose/xio/http/DefaultFullRequest.java (hunk 1 above)
Comment (manimaul) [reply xio-core/src/main/java/com/xjeffrose/xio/http/DefaultFullRequest.java]: done

File: xio-core/src/main/java/com/xjeffrose/xio/http/DefaultStreamingRequest.java (hunk 2 above)
Comment (manimaul) [reply xio-core/src/main/java/com/xjeffrose/xio/http/DefaultStreamingRequest.java]: done

**Pull request merged**